### ISSUE TITLE
diesel_cli on crates.io is outdated

### ISSUE THREAD

**Issue by ticky** [diesel-rs/diesel#2355]
### Versions

- **Rust:** rustc 1.41.1 (f3e1a954d 2020-02-24)
- **Diesel:** 1.4.0
- **Database:** sqlite
- **Operating System** macOS 10.14.6

### Feature Flags

- **diesel_cli:** postgres,sqlite,barrel-migrations

## Problem Description

The [latest `diesel_cli` on crates.io is 1.4.0](https://crates.io/crates/diesel_cli), which was published in early 2019. There appear to have been several updates to the CLI as well as dependencies since then, and with current rust versions it’s not possible to compile some of that version’s dependencies. It looks like this error should’ve been fixed by 15fc32b3040d222d5d54aba7ea35bbd344961777, but this commit (as well as other subsequent changes!) has not made it to crates.io!

### What are you trying to accomplish?

Install the `diesel_cli` with `barrel-migrations` enabled

### What is the expected output?

A lot of package downloads and builds followed by a 0 exit code

### What is the actual output?

A lot of package downloads followed by these compile errors:

```
error[E0557]: feature has been removed
  --> /Users/ticky/.cargo/registry/src/github.com-1ecc6299db9ec823/barrel-0.2.0/src/lib.rs:70:12
   |
70 | #![feature(custom_attribute)]
   |            ^^^^^^^^^^^^^^^^ feature has been removed
   |
   = note: removed in favor of `#![register_tool]` and `#![register_attr]`

error: cannot find attribute `rustfmt_skip` in this scope
  --> /Users/ticky/.cargo/registry/src/github.com-1ecc6299db9ec823/barrel-0.2.0/src/backend/pg.rs:38:11
   |
38 |         #[rustfmt_skip] /* This shouldn't be formatted. It's too long */
   |           ^^^^^^^^^^^^

error[E0554]: `#![feature]` may not be used on the stable release channel
  --> /Users/ticky/.cargo/registry/src/github.com-1ecc6299db9ec823/barrel-0.2.0/src/lib.rs:68:1
   |
68 | #![feature(box_syntax)]
   | ^^^^^^^^^^^^^^^^^^^^^^^

error[E0554]: `#![feature]` may not be used on the stable release channel
  --> /Users/ticky/.cargo/registry/src/github.com-1ecc6299db9ec823/barrel-0.2.0/src/lib.rs:69:1
   |
69 | #![feature(clone_closures)]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^

error[E0554]: `#![feature]` may not be used on the stable release channel
  --> /Users/ticky/.cargo/registry/src/github.com-1ecc6299db9ec823/barrel-0.2.0/src/lib.rs:70:1
   |
70 | #![feature(custom_attribute)]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error[E0308]: mismatched types
   --> /Users/ticky/.cargo/registry/src/github.com-1ecc6299db9ec823/barrel-0.6.5/src/integrations/diesel.rs:190:26
    |
190 | fn get_backend_pair() -> (&'static str, &'static str) {
    |    ----------------      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected tuple, found `()`
    |    |
    |    implicitly returns `()` as its body has no tail or `return` expression
    |
    = note:  expected tuple `(&'static str, &'static str)`
            found unit type `()`

error: aborting due to 5 previous errors

Some errors have detailed explanations: E0554, E0557.
For more information about an error, try `rustc --explain E0554`.
error: could not compile `barrel`.
warning: build failed, waiting for other jobs to finish...
error: aborting due to previous error

For more information about this error, try `rustc --explain E0308`.
error: could not compile `barrel`.
warning: build failed, waiting for other jobs to finish...
error: failed to compile `diesel_cli v1.4.0`, intermediate artifacts can be found at `/var/folders/x5/6y371x7n2rbbt_srkl7_m3mc0000gn/T/cargo-installetX5Mc`

Caused by:
  build failed
```

### Steps to reproduce

Run `cargo install diesel_cli --force --no-default-features --features postgres,sqlite,barrel-migrations`

## Checklist

- [x] I have already looked over the [issue tracker](https://github.com/diesel-rs/diesel/issues) for similar issues.
- [x] This issue can be reproduced on Rust's stable channel. (Your issue will be
  closed if this is not the case)

**Comment by weiznich** [comment]
We consider barrel related code to be part of the barrel project. This means we, as diesel maintainers,  won't actively backport anything. If anyone is interested in this feel free to open a PR against the appropriated branch, maybe someone finds time to issue a new release of diesel cli.

Otherwise: Yes that release from early 2019 is the latest release, as it matches the current stable diesel version. The next regular diesel_cli release will be done with the release of diesel 2.0 where we do explicitly not give any ETA's other then: "When it's done".

**Comment by ticky** [comment]
Does that mean that diesel 1.4.1 through 1.4.4 (current on crates.io) are not considered stable releases?

The code doesn’t appear to need updating, just releasing; the updated dependency is even [in the 1.4.x branch](https://github.com/diesel-rs/diesel/blob/1.4.x/diesel_cli/Cargo.toml) - the work seems to have already been done.

**Comment by weiznich** [comment]
> Does that mean that diesel 1.4.1 through 1.4.4 (current on crates.io) are not considered stable releases?

That does mean diesel 1.4.1 to 1.4.4 are patch releases, which implies that they contain specific minor bug fixes (as listed in the changelog). Non of those bug fixes affected `diesel_cli` so there was no need (and request) to issue a new release, as the old one remains compatible with all 1.4.x releases.

> The code doesn’t appear to need updating, just releasing; the updated dependency is even in the 1.4.x branch - the work seems to have already been done.

Seems like I've missed that at the time those commit were added to the release branch. If I find some spare time in the next week I will try issuing a patch release with the bumped version.

**Comment by weiznich** [comment]
I've published diesel_cli 1.4.1 to address this
